rules/mmseqs.smk

```diff
@@ -36,44 +36,161 @@ rule profile_to_pssm:
+    shell:
+        """
+
+            mmseqs search {input.fasta} {input.profile} {output.db} {config[tmpdir]}
+
+            mmseqs convertalis {input.fasta} {input.profile} {output.db} {output.tsv}
+
+        """
+
+## this rule works also for hmmer
+def get_all_targets_but(INPUT_TARGETS,remove_this):
+
+    return [ x for x in INPUT_TARGETS if not x==remove_this ]
+
+from Bio import SeqIO
+import numpy as np
+
+def subsample_fasta(input_fasta, N, output_fasta= None):
+    """
+        Subsample sequences in file. If N > than n seqs in file all sequences are retreved.
+        If outputfile is not specified, a list of BioPython Seqs are given.
+    """
+    seqs = list(SeqIO.parse(input_fasta,'fasta'))
+
+    if N >= len(seqs):
+        outseqs = seqs
+    else:
+        # subsample
+        outseqs= [seqs[j] for j in np.random.randint(0,len(seqs), N)]
+
+    if output_fasta is None:
+        return outseqs
+    else:
+        SeqIO.write(outseqs,output_fasta,'fasta')
+
+
+
+rule get_negative_evaluation_fasta:
+    input:
+        fasta = lambda wc: expand("{in_dir}/{targets}.{suffix}",
+                          targets = get_all_targets_but(INPUT_TARGETS, wc.input_targets),
+                          in_dir=config['in_dir'],suffix=config['suffix'])
+    output:
+        fasta= 'mmseqs/evaluation_seq/{input_targets}.negative.fasta',
+    params:
+        n_negatives = 2000
+    run:
+
+
+        if len(input.fasta) < 10:
+            shell("cat {input.fasta} > {output.fasta}")
+        else:
+            #subsample
+
+            # random int +- 1
+            n_samples_per_file = lambda : int(np.random.randn(1) + params.n_negatives / len(input.fasta) )
+
+            selected_sequences =[]
+
+            i=0
+            while len(selected_sequences) <= params.n_negatives:
+
+            # this acctually may subsample several times the same sequence
+
+                # cycle trough input files
+                seq_file = input.fasta[i % len(input.fasta) ]
+
+                i+=1
+                n_select= min([n_samples_per_file(), params.n_negatives-len(selected_sequences) ])
+
+                selected_sequences += subsample_fasta(seq_file, n_select)
+
+            assert len(selected_sequences) == params.n_negatives
+
+            SeqIO.write(selected_sequences,output.fasta,'fasta')
+
+
+rule score_mmseqs:
+    input:
+        profile = rules.MSAdb_to_profile.output,
+        fasta = 'mmseqs/evaluation_seq/{input_targets}.{classify_group}.db'
+    output:
+        db = temp('mmseqs/scores/{classify_group}/{input_targets}.scores'),
+        index = temp('mmseqs/scores/{classify_group}/{input_targets}.scores.index'),
+        tsv = 'mmseqs/scores/{classify_group}/{input_targets}.m8',
+    conda: "../envs/mmseqs.yaml"
     shell:
-        'mmseqs search {input.in2} {input.in1} {output.out1} {output.tmp}'
+        """
+
+        mmseqs search -e 1e10  {input.fasta} {input.profile} {output.db} {config[tmpdir]}
```

Comment:
We should use a permissive e-value like `-e 10` not `-e 1e10` because we want to capture all the distribution bitscores, not just the good hits (FYI: i still have not got the tests working yet)